src/classes/CON_Phone_Batch.cls

```diff
@@ -0,0 +1,137 @@
+/*
+    Copyright (c) 2019, Salesforce.org Foundation
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of the Salesforce.org Foundation nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.com Foundation
+* @date 2019
+* @group Contacts
+* @group-content ../../ApexDocContent/Contacts.htm
+* @description Batch Class to process contacts for Preferred Phone when run from EDA Settings
+*/
+public class CON_Phone_Batch implements Database.Batchable<sObject>,Database.Stateful {
+    public List<String> exception_List = new List<String>();
+    public Integer recordCount = 0;
+  
+    String query = 'SELECT Id, Name, Phone, PreferredPhone__c '+getPhoneFields()+' FROM Contact';
+
+    private InstallContext context;
+
+    public CON_Phone_BATCH(InstallContext ic) {
```

Comment:
@skosgi1 Last change before I move this onward, I promise! I don't think we need this parameter or constructor. So lines 43-47 can be removed. (This will require updates to classes and test classes that instantiate this class, but hopefully shouldn't be too significant as I see we're just passing in "null" everywhere.)

Reply:
@robertbwatson Looks like context is used on line 102, to get the id of user and send email.

Reply:
Oh, interesting - I'm really curious about the testing of this functionality. The context is literally always null - we don't instantiate this through the install script, so we're always passing in null, at least from what I could tell from reviewing the pull request, thus that block of code will never run either.

Reply:
We can keep it as is if we want to keep this moving forward. But if so, we need to create a tech debt ticket to handle removing unused code.

Reply:
@skosgi1 If you create a tech debt ticket to go through both the Preferred Phone and Preferred Email functionality to remove unused code and refactor the logic where possible, I think we can move this PR forward.

Reply:
@robertbwatson created W-034768.